sr_utilities_common/test/aws_manager_test.py

```diff
@@ -14,49 +14,118 @@
-                                'test', ["test.txt"])
-        folder_list = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, 'test')
+    def test_aws_02_upload_subfolder(self):
+        self.filename = get_filename()  # Second file should have new filename
+        result = False
+        self.make_files_to_be_uploaded()
+        self.aws_manager.upload(BUCKET_NAME, "/tmp", "upload", [self.filename], "Subfolder")
+        folder_list = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, None)
         if folder_list is not None:
             for element in folder_list:
-                if element['Key'] == 'test/test.txt':
-                    return True
-        return False
+                if element['Key'] == f"Subfolder/{self.filename}":
+                    result = True
+        self.assertTrue(result)
 
+    def test_aws_03_download(self):
+        result = False
+        delete_download_folder()
+        self.aws_manager.download(BUCKET_NAME, "/tmp", "download", [self.filename], "upload")
+        if os.path.exists(f"{DOWNLOAD_PATH}/{self.filename}"):
+            result = True
+        self.assertTrue(result)
 
-if __name__ == '__main__':
-    rospy.init_node('aws_manager_test')
+    def test_aws_04_download_subfolder(self):
+        result = False
+        self.aws_manager.download(BUCKET_NAME, "/tmp", "download", [self.filename], "Subfolder")
+        if os.path.exists(f"{DOWNLOAD_PATH}/{self.filename}"):
+            result = True
+        self.assertTrue(result)
+
+    def test_aws_05_remove_all_files(self):
+        rospy.sleep(4)
+        folder_list = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, None)
+        if folder_list is not None:
+            for element in folder_list:
+                filepath = element['Key']
+                rospy.loginfo(filepath)
+                self.aws_manager._client.delete_object(Bucket=BUCKET_NAME, Key=filepath)
+        # Check bucket status after removing everything.
+        folder_list = self.aws_manager.get_bucket_structure_with_prefix(BUCKET_NAME, None)
+        if not folder_list:
+            result = True
+        else:
+            result = False
+        self.assertTrue(result)
 
-    test_aws_manager = Test_AWS_Manager()
-    method_list = ['aws_get_bucket_structure_with_prefix',
-                   'aws_download', 'aws_upload']
-    test_suite = ManualTestSuite(test_aws_manager, method_list)
+    def make_files_to_be_uploaded(self):
+        if not os.path.exists(UPLOAD_PATH):
+            os.mkdir(UPLOAD_PATH)
+        with open(f"{UPLOAD_PATH}/{self.filename}", "w") as upload_file:
+            letters = string.ascii_lowercase
+            message = ''.join(random.choice(letters) for i in range(10))
+            upload_file.write(message)
+
+
+def get_filename():
+    time = datetime.now()
+    current_time = time.strftime("%d%m%Y_%H.%M.%S")
+    return f"test_{current_time}"
```

Comment:
If we don't store for a longer period the files in AWS, or don't check anything time related maybe you could remove the datetime import and name the file with some random string?

Reply:
Good suggestion! will add